Source/WebCore/page/Navigator.cpp

```diff
@@ -265,11 +324,14 @@ void Navigator::initializePluginAndMimeTypeArrays()
         additionalWebVisibleDOMPlugins.uncheckedAppend(WTFMove(wrapper));
     }
 
-    std::sort(publiclyVisibleDOMPlugins.begin(), publiclyVisibleDOMPlugins.end(), [](const Ref<DOMPlugin>& a, const Ref<DOMPlugin>& b) {
+    ASSERT(!m_pdfViewerEnabled || publiclyVisibleDOMPlugins.begin()->get().info().name == genericPDFViewerName);
+    size_t dummyPDFOffset = m_pdfViewerEnabled ? 1 : 0;
+    std::sort(publiclyVisibleDOMPlugins.begin() + dummyPDFOffset, publiclyVisibleDOMPlugins.end(), [](const Ref<DOMPlugin>& a, const Ref<DOMPlugin>& b) {
```

Comment:
Why this sorting? And why do we have additionalWebVisiblePlugins?

Per the spec, the list should either be empty (when pdf is not supported) or a dummy static list of 5 PDF plugins (when pdf is supported).

It doesn't look like we are really matching the spec here? Our logic looks terribly complicated for something that looks so simple in the spec.

Reply:
All of that was a holdover from when we had many plugins, and wanted to always report the plugins in the same order regardless of when the user installed a plugin, or enabled/disabled them. This is no longer needed so I removed it in a follow-up.